src/lib/network_pool/indexed_pool.ml

```diff
@@ -387,6 +401,27 @@ let remove_applicable_exn :
     applicable_by_fee = Map_set.remove_exn t.applicable_by_fee fee_per_wu cmd
   }
 
+(* Remove a command from entry_times and all_by_entry_time *)
+let remove_entry_times (cmd_hash : Transaction_hash.t) (t : t) : t =
+  let entry_time = Map.find t.entry_times cmd_hash in
+  match entry_time with
+  | Some entry_time ->
+      { t with
+        entry_times = Map.remove t.entry_times cmd_hash
+      ; all_by_entry_time = Map.remove t.all_by_entry_time entry_time
+      }
+  | None ->
+      t
```

Comment:
Doesn't this represent a broken invariant? It seems like this should be an exception, to make sure that we aren't allowing transactions in that don't have a removal time.

Reply:
Done!

Reply:
You may need to push, it looks unchanged here.